packages/web-components/fast-foundation/src/tree-item/tree-item.ts

```diff
@@ -0,0 +1,269 @@
+import {
+    attr,
+    FASTElement,
+    Notifier,
+    Observable,
+    observable,
+} from "@microsoft/fast-element";
+import {
+    getDisplayedNodes,
+    isHTMLElement,
+    keyCodeArrowDown,
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeArrowUp,
+    keyCodeEnter,
+    keyCodeSpace,
+} from "@microsoft/fast-web-utilities";
+
+export class TreeItem extends FASTElement {
+    @attr({ mode: "boolean" })
+    public expanded: boolean = false;
+
+    @attr({ mode: "boolean" })
+    public selected: boolean;
+
+    @attr({ mode: "boolean" })
+    public disabled: boolean;
+
+    public expandCollapseButton: HTMLDivElement;
+
+    public treeItem: HTMLElement;
+
+    @observable
+    public focusable: boolean = false;
+
+    public afterContent: HTMLSlotElement;
```

Comment:
See above note about start/end - you can use the mixin like button/anchor to apply the functionality

Reply:
oh, I still need to remove these after adding the template pattern